airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/context/env/ConnectorConfigurationPropertySource.kt

```diff
@@ -0,0 +1,155 @@
+            return values
+        }
+
+        private fun loadFile(
+            propertyFilePath: String?,
+            prefix: String,
+        ): Map<String?, Any?> {
+            if (StringUtils.hasText(propertyFilePath)) {
+                val propertyFile = propertyFilePath?.let { Path.of(it).toFile() }
+                if (propertyFile?.exists() == true) {
+                    val properties: Map<String?, Any?>? =
+                        Jsons.deserialize(propertyFile.readText(), MapTypeReference())
+                    return flatten(properties!!, prefix)
+                        .collect(
+                            Collectors.toMap(
+                                { e: Map.Entry<String?, Any?> -> e.key },
+                                { e: Map.Entry<String?, Any?> -> e.value },
+                            ),
+                        )
+                } else {
+                    logger.warn { "Property file '$propertyFile', not found for prefix '$prefix'." }
+                    return mapOf()
+                }
+            } else {
+                return mapOf()
+            }
+        }
+
+        private fun loadFileContents(
+            propertyFilePath: String?,
+            prefix: String,
+        ): Map<String?, Any?> {
+            if (StringUtils.hasText(propertyFilePath)) {
+                val propertyFile = propertyFilePath?.let { Path.of(it).toFile() }
+                if (propertyFile?.exists() == true) {
+                    return java.util.Map.of<String, Any>(prefix, propertyFile.readText())
+                } else {
+                    logger.warn { "Property file '$propertyFile', not found for prefix '$prefix'." }
+                    return mapOf()
+                }
+            } else {
+                return mapOf()
+            }
+        }
+
+        private fun flatten(
+            map: Map<String?, Any?>,
+            prefix: String,
+        ): Stream<Map.Entry<String?, Any?>> {
+            return map.entries.stream().flatMap { e: Map.Entry<String?, Any?> ->
+                flattenValue(
+                    e,
+                    prefix,
+                )
+            }
+        }
+
+        private fun flattenValue(
+            entry: Map.Entry<String?, Any?>,
+            prefix: String,
+        ): Stream<Map.Entry<String?, Any?>> {
+            return if (entry.value is Map<*, *>) {
+                flatten(
+                    entry.value as Map<String?, Any?>,
+                    String.format(PREFIX_FORMAT, prefix, entry.key)
+                )
+            } else {
+                Stream.of(
+                    AbstractMap.SimpleEntry(
+                        String.format(
+                            PREFIX_FORMAT,
+                            prefix,
+                            entry.key,
+                        ),
+                        entry.value,
+                    ),
+                )
+            }
+        }
+    }
```

Comment:
I'll be honest I found this companion object code hard to read and just glossed over it. Consider improving it.